resources/codegen/tests/test___init__.py

```diff
@@ -0,0 +1,318 @@
+        a_capturing_logger,
+    ):
+        openassetio_codegen.generate(
+            description_path=yaml_path_all,
+            output_directory=some_output_dir,
+            languages=["a"],
+            creation_callback=some_creation_callback,
+            logger=a_capturing_logger,
+            dry_run=False,
+        )
+
+        mock_language_a.generate.assert_called()
+
+    def test_when_no_globals_then_generator_called_with_default_globals(
+        self,
+        yaml_path_all,
+        some_output_dir,
+        some_creation_callback,
+        mock_language_a,
+        a_capturing_logger,
+    ):
+
+        openassetio_codegen.generate(
+            description_path=yaml_path_all,
+            output_directory=some_output_dir,
+            languages=["a"],
+            creation_callback=some_creation_callback,
+            logger=a_capturing_logger,
+        )
+
+        expected_globals = openassetio_codegen.generators.helpers.default_template_globals()
+        expected_globals["language"] = "a"
+
+        call_args = mock_language_a.generate.call_args[0]
+        assert call_args[1] == expected_globals
+
+    def test_when_globals_supplied_then_generator_called_with_updated_globals(
+        self,
+        yaml_path_all,
+        some_output_dir,
+        some_creation_callback,
+        mock_language_a,
+        a_capturing_logger,
+    ):
+
+        extra_globals = {"a": 1, "b": "🤠", "copyrightOwner": "Me"}
+
+        openassetio_codegen.generate(
+            description_path=yaml_path_all,
+            output_directory=some_output_dir,
+            languages=["a"],
+            creation_callback=some_creation_callback,
+            logger=a_capturing_logger,
+            template_globals=extra_globals,
+        )
+
+        expected_globals = openassetio_codegen.generators.helpers.default_template_globals()
+        expected_globals.update(extra_globals)
+        expected_globals["language"] = "a"
+
+        call_args = mock_language_a.generate.call_args[0]
+        assert call_args[1] == expected_globals
+
+    def test_when_globals_supplied_include_language_then_generator_global_not_overridden(
+        self,
+        yaml_path_all,
+        some_output_dir,
+        some_creation_callback,
+        mock_language_a,
+        a_capturing_logger,
+    ):
+
+        extra_globals = {"language": "b"}
+
+        openassetio_codegen.generate(
+            description_path=yaml_path_all,
+            output_directory=some_output_dir,
+            languages=["a"],
+            creation_callback=some_creation_callback,
+            logger=a_capturing_logger,
+            template_globals=extra_globals,
+        )
+
+        expected_globals = openassetio_codegen.generators.helpers.default_template_globals()
+        expected_globals["language"] = "a"
+
+        call_args = mock_language_a.generate.call_args[0]
+        assert call_args[1] == expected_globals
+
+    def test_when_dry_run_set_true_then_structure_is_logged_as_info(
```

Comment:
[question] This implied (to me) that structure is logged _only_ if doing a dry run.  Was that the intention, or was the test named this way just to ensure that dry-run didn't skip logging?

Reply:
Renamed [here](https://github.com/OpenAssetIO/OpenAssetIO/pull/529/commits/3b016992912a4e1955e3caf1e36ef32ae4e0c81a).

Reply:
That's clearer, thanks.  A minor point, but do we then need a test that the structure is logged when not doing a dry-run?

Reply:
Ahh yes! Added [here](https://github.com/OpenAssetIO/OpenAssetIO/pull/529/commits/164bf48cf1c9fadfd1fb9d2c52f98e20279fe526).

Reply:
Thanks. Missing a "then" in the test case name though :grin:

Reply:
DOH! ([added](https://github.com/OpenAssetIO/OpenAssetIO/pull/529/commits/c3bf27f3b0dfd33744c0d7cd2eefc0d29209c8c9))